actionview/lib/action_view/renderer/partial_renderer.rb

```diff
@@ -405,7 +405,11 @@ def setup(context, options, block)
     def collection_from_options
       if @options.key?(:collection)
         collection = @options[:collection]
-        collection.respond_to?(:to_ary) ? collection.to_ary : []
+        if collection.respond_to?(:to_ary)
+          collection = collection.to_ary
+        else
+          raise ArgumentError.new("'#{collection.inspect}' is not an ActiveModel-compatible object. It must implement :to_ary.")
```

Comment:
This error is misleading; ActiveModel doesn't have any expectation of a `to_ary` method.

I'm also hesitant about including the inspect output, but if we do, it shouldn't be quoted like this.